### PR TITLE
Changes for Xcode 8 beta 4

### PR DESCRIPTION
**Pull request by jameshurst**

### CODE REVIEW

**Comment by mdiep** [comment]
Thanks for working on this!

It looks like the unit tests need to be updated as well.

**Comment by jameshurst** [comment]
Oops, I guess I hadn't run the unit tests and missed that those needed to be updated. They should be all fixed now.

**Comment by mdiep** [comment]
Looks good. 👍🏻 But we need Travis to update so the tests pass.

**Comment by NachoSoto** [comment]
> But we need Travis to update so the tests pass.

See you in 2017.

**Comment by mdiep** [comment]
Would you mind updating the `.swift-version` file to point to a compatible Linux release so those tests pass on CI? This is ready to go apart from that. (Travis updated their image tonight.)

**Comment by jameshurst** [comment]
Alright, just updated the `.swift-version` file to `3.0-PREVIEW-3`. Lets hope the tests pass in Tavis.

**Comment by mdiep** [comment]
It looks like we need to go [to a snapshot](https://github.com/antitypical/Result/pull/176). If you're not up for tackling this, we can probably do it separately.

**Comment by jameshurst** [comment]
Ah yep, it looks like preview 3 doesn't have those source changes included in the version that ships with Xcode 8 beta 4. I'll set the `.swift-version` to the latest development snapshot.

**Comment by jameshurst** [comment]
It looks like there were some changes to how tests get handled in the latest development snapshot.

```
error: the module at Tests/Result has an invalid name ('Result'): the name of a test module has no ‘Tests’ suffix
fix: rename the module at ‘Tests/Result’ to have a ‘Tests’ suffix
```

**Comment by NachoSoto** [comment]
Re-opening PR to trigger tests. They should pass now that Travis has beta 4 :)

**Comment by NachoSoto** [comment]
Tests seem to have [failed in Linux](https://travis-ci.org/antitypical/Result/jobs/149731613), the rest passed.

**Comment by jameshurst** [comment]
Yeah, I tried to get the tests fixed on Linux yesterday but got stuck on some linker errors while building the tests. Perhaps someone more familiar with SPM would be able to take a look.

**Comment by norio-nomura** [comment]
`swift-3.0-PREVIEW-4` has come https://swift.org/download/#previews

**Comment by kylef** [comment]
```
error: the module at Tests/Result has an invalid name ('Result'): the name of a test module has no ‘Tests’ suffix
fix: rename the module at ‘Tests/Result’ to have a ‘Tests’ suffix
```

I think this error is saying `Tests/Result` should be renamed to `Tests/ResultTests`?

File: .swift-version [hunk 1]
```diff
@@ -1 +1 @@
-3.0-PREVIEW-2
+DEVELOPMENT-SNAPSHOT-2016-07-29-a
```
Comment (ikesyo) [inline .swift-version]: Let's use `3.0-PREVIEW-4`: https://github.com/antitypical/Result/pull/178#issuecomment-238004406

**Comment by jameshurst** [comment]
It looks like with `3.0-PREVIEW-4` the change to rename `Test/Result` to have a `Tests` suffix isn't needed.

**Comment by jameshurst** [comment]
Looks like everything has passed in Travis.

**Comment by mdiep** [comment]
This looks great! Thanks for all the work! :sparkles:

**Pull request merged**